src/App.tsx

```diff
@@ -1,61 +1,110 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { Todo, User } from './react-app-env';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todosWithUser, setTodosWithUser] = useState<Todo[]>(todos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserById, setSelectedUserById] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const resetForm = () => {
+    setTodoTitle('');
+    setSelectedUserById(0);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!todoTitle);
+    setHasUserIdError(!selectedUserById);
+
+    const user = getUser(selectedUserById);
+
+    const newId = Math.max(...todosWithUser.map(todo => todo.id)) + 1;
```

Comment:
What you do now:
1. Iterating each todo into its id
2. Iterating each id to find the max value
3. Adding a one to it.

Why didn't you just use .length + 1?

Reply:
But the id's, are kinda random there can't use length. Or did I misunderstand you?

Reply:
I agree 👍